### PR TITLE
Landing Page Type

### PR DESCRIPTION
**Pull request by kave**
## Additions

- New Landing Page type for wagtail

## Changes

- Routed landing page type related molecules and organisms to use `jinja2/v1/_includes/` template locations

## Testing

- `pip install -r requirements/base.txt`
- Create landing page type and fill out content.

## Review

- @kurtw 
- @rosskarchner 
- @jimmynotjim 
- @anselmbradford 
- @KimberlyMunoz 
- @sebworks

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -91,6 +91,7 @@ Given the `MAJOR.MINOR.PATCH` pattern, here is how we decide to increment:
 - Added Call to Action backend and template
 - Added Contact snippet and molecule backends
 - Added temporary folder for converted Jinja2 Wagtail field template files
+- Added WP Import Data Contact processor
 - Added templates and CSS for the Adding Sidebar Breakout organism.
```
Comment (kave) [inline CHANGELOG.md]: -_- not sure how this snuck in here

File: cfgov/jinja2/v1/_includes/molecules/hero.html [hunk 2]
```diff
@@ -34,31 +34,65 @@
 
    ========================================================================== #}
 
-{% macro render( settings, options={} ) %}
-
-{% if settings.heading and settings.body and settings.image_url %}
+{% if value %}
```
Comment (kave) [inline cfgov/jinja2/v1/_includes/molecules/hero.html]: I choose to edit the templates directly instead of creating a new file because it would seem redundant to maintain two separate copies. this will lead to the `jinja2\v1\wagtail` directory eventually being deleted

File: cfgov/v1/__init__.py [hunk 3]
```diff
@@ -15,9 +16,25 @@ def environment(**options):
...
     })
     env.filters.update({
         'slugify': slugify,
         'richtext': wagtailcore_tags.richtext
     })
     return env
+
+
+@contextfunction
+def render_stream_child(context, stream_child):
```
Comment (sebworks) [inline cfgov/v1/__init__.py]: Is this function necessary to render Jinja2 templates in Wagtail?

Comment (kave) [inline cfgov/v1/__init__.py]: its necessary to specifically to streamblocks with richtext field. without it, it prints those stream fields as raw html instead of rendering them.

Comment (sebworks) [inline cfgov/v1/__init__.py]: awesome, thanks

Comment (kave) [inline cfgov/v1/__init__.py]: yea i'd like to get rid of this eventually, hopefully they will implement a cleaner solution in the next release

File: cfgov/v1/models/molecules.py [hunk 4]
```diff
@@ -79,7 +79,7 @@ class ImageText2575(blocks.StructBlock):
     def clean(self, data):
         error_dict = {}
         try:
-            block_data = super(ImageText5050, self).clean(data)
+            block_data = super(ImageText2575, self).clean(data)
```
Comment (kurtw) [inline cfgov/v1/models/molecules.py]: lol this was probably my handy work.

File: cfgov/v1/models/molecules.py [hunk 5]
```diff
@@ -117,16 +117,76 @@ class TextIntroduction(blocks.StructBlock):
...
+
+class Hero(blocks.StructBlock):
+    heading = blocks.CharBlock(max_length=100, required=True)
+    body = blocks.RichTextBlock(required=False)
+
+    image = ImageChooserBlock(required=False)
+    image_path = blocks.CharBlock(required=False)
+    image_alt = blocks.CharBlock(required=False)
+
+    background_color = blocks.CharBlock(max_length=100, required=False)
```
Comment (kurtw) [inline cfgov/v1/models/molecules.py]: Not sure if this is something we should be handling in the CMS or not. @sebworks @anselmbradford @jimmynotjim @KimberlyMunoz how is background determined in the front end? If there's a manageable set of background colors to use then we can set up a ChoiceField to select from that list if it makes sense to do so. What do yall think?

File: cfgov/jinja2/v1/_includes/molecules/hero.html [hunk 6]
```diff
@@ -34,31 +34,65 @@
...
-            <p class="hero_subhead">{{ settings.body }}</p>
-            {% if options.link_text and options.link_url %}
-                <a href="{{ options.link_url }}"
+            <h1 class="hero_heading">{{ value.heading | safe}}</h1>
+            <p class="hero_subhead">{{ value.body |richtext}}</p>
+            {% if value.link_text and value.link_url %}
+                <a href="{{ value.link_url }}"
                    class="hero_cta
-                   {{ 'btn' if options.is_button
+                   {{ 'btn' if value.is_button
```
Comment (kurtw) [inline cfgov/jinja2/v1/_includes/molecules/hero.html]: I'm not seeing the button in the designs. @anselmbradford?

File: cfgov/v1/models/molecules.py [hunk 7]
```diff
@@ -117,16 +117,76 @@ class TextIntroduction(blocks.StructBlock):
...
+
+        if block_data['image'] and block_data['image_path']:
+            img_err = ['Please select one method of image rendering']
+            error_dict.update({
+                'image': img_err,
+                'image_path': img_err})
+
+        if block_data['image_path'] and not block_data['image_alt']:
+            error_dict.update({'image_alt': isRequired('Image Alt')})
+
```
Comment (kurtw) [inline cfgov/v1/models/molecules.py]: I think there should also be a validation error if there's an `image_alt` but not an `image_path`.

Comment (kave) [inline cfgov/v1/models/molecules.py]: True, i will add that in the next PR where i've extracted this logic out into the atoms model file

Comment (kurtw) [inline cfgov/v1/models/molecules.py]: :+1:

File: cfgov/v1/models/molecules.py (hunk 5 above)
Comment (jimmynotjim) [inline cfgov/v1/models/molecules.py]: I think that's actually a question for designers. I know there's at least a few colors they wanted to choose from, maybe we make it a dropdown? cc @ajbush @duelj

File: cfgov/jinja2/v1/_includes/molecules/hero.html (hunk 6 above)
Comment (jimmynotjim) [inline cfgov/jinja2/v1/_includes/molecules/hero.html]: It's in the [Design Manual Spec](http://stephanieosan.github.io/design-manual/ui-toolkit/heroes.html) as "Call to Action Button"

File: cfgov/v1/models/molecules.py [hunk 8]
```diff
@@ -117,16 +117,77 @@ class TextIntroduction(blocks.StructBlock):
...
+    class Meta:
+        icon = 'image'
+        template = '_includes/molecules/hero.html'
+
+
+class FormFieldWithButton(blocks.StructBlock):
+    btn_text = blocks.CharBlock(max_length=100, required=True)
+
+    required = blocks.BooleanBlock(required=False)
+    id = blocks.CharBlock(max_length=100, required=False)
```
Comment (kurtw) [inline cfgov/v1/models/molecules.py]: Possibly remove.

File: cfgov/v1/models/molecules.py (hunk 7 above)
Comment (jimmynotjim) [inline cfgov/v1/models/molecules.py]: Hero images should always be background images and thus don't need an `alt` attribute

File: cfgov/v1/models/molecules.py [hunk 9]
```diff
@@ -117,16 +117,77 @@ class TextIntroduction(blocks.StructBlock):
...
+    info = blocks.RichTextBlock(required=False)
+    label = blocks.CharBlock(max_length=100, required=True)
+    type = blocks.CharBlock(max_length=100, required=True)
```
Comment (kurtw) [inline cfgov/v1/models/molecules.py]: Now I'm wondering if this, and other fields, can be passed in when the block is declared.

For example from the Email org: `form_field = blocks.ListBlock(molecules.FormFieldWithButton(type='email'), icon='mail', required=False)`

File: cfgov/jinja2/v1/landing-page/index.html [hunk 10]
```diff
@@ -14,129 +14,54 @@
...
-                'link_text' :   'Visit Google'
-            }) }}
+        {% for block in page.content %}
+            {% if block.block_type == 'image_text_50_50' %}
+                {{ render_stream_child(block) }}
+            {% endif %}
+        {% endfor %}
     </div>
     <div class="block">
         <h2>Headline for 2/7 organism</h2>
```
Comment (kave) [inline cfgov/jinja2/v1/landing-page/index.html]: @KimberlyMunoz hey can this heading be removed?

File: cfgov/jinja2/v1/_includes/molecules/hero.html (hunk 6 above)
Comment (kurtw) [inline cfgov/jinja2/v1/_includes/molecules/hero.html]: Right, but I'm not seeing that in the Hero specs. It's usually called CTA standalone w/ button right?

File: cfgov/v1/models/molecules.py [hunk 11]
```diff
@@ -117,16 +117,77 @@ class TextIntroduction(blocks.StructBlock):
...
+        except ValidationError as e:
+            error_dict.update(e.params)
+            block_data = data
+
+        if not block_data['image'] and not block_data['image_path'] and not block_data['image_alt']:
+            img_err = ['Please upload or enter an image path']
+            error_dict.update({'image': img_err, 'image_path': img_err, 'image_alt': isRequired('Image alt')})
+
+        if block_data['image'] and block_data['image_path']:
+            img_err = ['Please select one method of image rendering']
```
Comment (jimmynotjim) [inline cfgov/v1/models/molecules.py]: What are the methods here? Hero images will always be background images.

File: cfgov/jinja2/v1/_includes/molecules/form-field-with-button.html [hunk 12]
```diff
@@ -27,24 +27,46 @@
...
+    <div class="m-form-field-with-button">
+        <div class="form-group">
+            <label for="{{ value.name }}">
+                <b>{{ value.label }}</b>
+                {% if value.required %}
+                    (required)
+                {% endif %}
+            </label>
+            <input id="{{ (value.id or value.name) ~
+                    '_' ~ range(1, 100) | random }}"
```
Comment (KimberlyMunoz) [inline cfgov/jinja2/v1/_includes/molecules/form-field-with-button.html]: Because of this _random-number, the input `id` doesn't match the label and we're getting an accessibility error. This should just be `<input id="{{ (value.id or value.name)">` and there should be something to prevent an id from being used more than once.

File: cfgov/jinja2/v1/_includes/molecules/form-field-with-button.html [hunk 13]
```diff
@@ -27,24 +27,46 @@
...
+        <div class="m-form-field-with-button">
+            <div class="form-group">
+                <label for="{{ settings.field.name }}">
+                    <b>{{ settings.field.label }}</b>
+                    {% if settings.field.required %}
+                        (required)
+                    {% endif %}
+                </label>
+                <input id="{{ (settings.field.id or settings.field.name) ~
+                        '_' ~ range(1, 100) | random }}"
```
Comment (KimberlyMunoz) [inline cfgov/jinja2/v1/_includes/molecules/form-field-with-button.html]: Same issue as above.

File: cfgov/jinja2/v1/landing-page/index.html (hunk 10 above)
Comment (KimberlyMunoz) [inline cfgov/jinja2/v1/landing-page/index.html]: Yes. We might have to add a way to add a title to a group like this. Should we make an atom macro for that?

Comment (jimmynotjim) [inline cfgov/jinja2/v1/landing-page/index.html]: I think there's supposed to always be a headline over a batch of 25/75 molecules cc @ajbush @duelj

File: cfgov/jinja2/v1/_includes/molecules/hero.html (hunk 6 above)
Comment (jimmynotjim) [inline cfgov/jinja2/v1/_includes/molecules/hero.html]: It's an option to turn the link into a button. Design just needs to update the v1 Hero spec.

File: cfgov/jinja2/v1/landing-page/index.html (hunk 10 above)
Comment (KimberlyMunoz) [inline cfgov/jinja2/v1/landing-page/index.html]: Do you think it would be okay to make an atom molecule for a group title? Or should we adjust the image-text molecule to take this batching into account?

Comment (jimmynotjim) [inline cfgov/jinja2/v1/landing-page/index.html]: I think an atom that can be used here and anywhere else we have a chunk of molecules is a great solution. cc @ajbush @duelj

File: cfgov/jinja2/v1/_includes/molecules/form-field-with-button.html (hunk 12 above)
Comment (sebworks) [inline cfgov/jinja2/v1/_includes/molecules/form-field-with-button.html]: Yep, this won't ensure uniqueness. Maybe use a context function which returns `uuid.uuid1().hex`?

**Comment by duelj** [comment]
I think I am following the conversation here and from the design end we would like a title for a group of molecules, like the 50/50 text blob, 25/75, etc. I think the atom idea for the title makes sense and is how we were viewing it from the design end as well. Would it help to have an organism for a group of molecules like this with a header or description paragraph?

File: cfgov/jinja2/v1/_includes/molecules/form-field-with-button.html (hunk 12 above)
Comment (KimberlyMunoz) [inline cfgov/jinja2/v1/_includes/molecules/form-field-with-button.html]: Before we try to find a solution on the front-end, is there something on the back-end that can detect if an ID is being used more than once in a form-field?

I'm not a big fan of using something like uuid, which is a little ~too~ unique and overkill for this situation, especially since our pages are not on HTTPS. But I'm open to it.

File: cfgov/jinja2/v1/_includes/molecules/hero.html (hunk 6 above)
Comment (kurtw) [inline cfgov/jinja2/v1/_includes/molecules/hero.html]: @ajbush @duelj

File: cfgov/v1/models/molecules.py (hunk 7 above)
Comment (kurtw) [inline cfgov/v1/models/molecules.py]: Thanks for mentioning that.

File: cfgov/v1/models/molecules.py (hunk 9 above)
Comment (kurtw) [inline cfgov/v1/models/molecules.py]: Via verbal agreement, we're changing this to a ChoiceBlock.

Comment (kave) [inline cfgov/v1/models/molecules.py]: i'll turn this into a chooser block instead with all the available input tag types

File: cfgov/v1/models/molecules.py (hunk 11 above)
Comment (kave) [inline cfgov/v1/models/molecules.py]: Yea correct, so abstracted this validation logic into the atom instead,so its not Hero specifc to so its reusable and opted to make `alt` not required

**Comment by kave** [comment]
@duelj So yes this could work. So the molecule would look like this right?
Organism:
- heading 
- body
- molecule plural of the same type

@KimberlyMunoz This way we wouldn't have to edit the molecules themselves

File: cfgov/v1/models/molecules.py [hunk 14]
```diff
@@ -12,121 +12,90 @@ def isRequired(field_name):
...
+
+    class Meta:
+        icon = 'mail'
+        template = '_includes/molecules/form-field-with-button.html'
 
 
 class CallToAction(blocks.StructBlock):
+    heading = blocks.CharBlock(max_length=100, required=True)
     slug = blocks.CharBlock(required=True)
-    paragraph = blocks.RichTextBlock()
```
Comment (kurtw) [inline cfgov/v1/models/molecules.py]: Why was this removed?

File: cfgov/v1/models/atoms.py [hunk 15]
```diff
@@ -1,6 +1,70 @@
...
+
+
+def isRequired(field_name):
+    return [str(field_name) + ' is required.']
 
 
 class Hyperlink(blocks.StructBlock):
-    label = blocks.CharBlock(max_length=50)
-    href = blocks.CharBlock(default='/')
+    text = blocks.CharBlock(max_length=50, required=False)
```
Comment (kurtw) [inline cfgov/v1/models/atoms.py]: Why were these names changed?

File: cfgov/jinja2/v1/_includes/molecules/form-field-with-button.html (hunk 12 above)
Comment (kurtw) [inline cfgov/jinja2/v1/_includes/molecules/form-field-with-button.html]: So these form field blocks will have to have unique id's across the page?

Comment (KimberlyMunoz) [inline cfgov/jinja2/v1/_includes/molecules/form-field-with-button.html]: Yep, and the id should match `<label for="ID">` for accessibility reasons.

File: cfgov/v1/models/atoms.py (hunk 15 above)
Comment (kave) [inline cfgov/v1/models/atoms.py]: to be consistent with the templates

File: cfgov/v1/models/molecules.py (hunk 14 above)
Comment (kave) [inline cfgov/v1/models/molecules.py]: to be consistent with the template

File: cfgov/v1/models/atoms.py (hunk 15 above)
Comment (kurtw) [inline cfgov/v1/models/atoms.py]: These fields were changed here but not in the templates. Either the templates should change or the fields should change.

Comment (kave) [inline cfgov/v1/models/atoms.py]: huh? they are all changed in the molecule templates that refer to links. can you point one out where your seeing the change not made.

File: cfgov/v1/models/molecules.py (hunk 14 above)
Comment (kurtw) [inline cfgov/v1/models/molecules.py]: @duelj @ajbush @jimmynotjim  [The specifications for the CTA](https://github.cfpb.gov/flapjack/Modules-V1/wiki/Standalone-CTA-\(with-button\)) say that this has paragraphs. Did this change or are the templates just not including those specs?

File: cfgov/v1/models/atoms.py (hunk 15 above)
Comment (kurtw) [inline cfgov/v1/models/atoms.py]: https://github.com/cfpb/cfgov-refresh/pull/1154/files#diff-3f1f0f338f0a28057bf1dac682d9dca2R43

Comment (sebworks) [inline cfgov/v1/models/atoms.py]: ok, so the problem is that `value.slug` should be `value.text`. I am referring to https://github.com/cfpb/cfgov-refresh/pull/1154/files#diff-3f1f0f338f0a28057bf1dac682d9dca2R36.

File: cfgov/v1/models/molecules.py (hunk 14 above)
Comment (duelj) [inline cfgov/v1/models/molecules.py]: We will want the option of paragraph text with the standalone CTA molecule. The lorem ipsum in [this image](https://github.cfpb.gov/github-enterprise-assets/0000/0160/0001/5470/69ca1838-6101-11e5-819d-7a9cc6be43e2.png) would represent how we would use it.

File: cfgov/jinja2/v1/_includes/molecules/hero.html [hunk 16]
```diff
@@ -34,31 +34,65 @@
...
                 </a>
             {% endif %}
         </div>
-        <div class="hero_image"
-             style="background-image: url( {{ settings.image_url }} );">
+        {% if value.image.upload %}
+            {% set photo=image(value.image.upload, 'original') %}
+            <div class="hero_image"
+                 style="background-image: url( {{ photo.url }} );">
+        {% else %}
```
Comment (KimberlyMunoz) [inline cfgov/jinja2/v1/_includes/molecules/hero.html]: When we talked with content earlier last week, they said they would just want the image upload and not the image url option. Since it isn't clear that you have to put in one or the other until you submit, can we remove the second option? @sarahsimpson09 

image

Comment (KimberlyMunoz) [inline cfgov/jinja2/v1/_includes/molecules/hero.html]: And we should be able to remove the alt since it's a background image.

File: cfgov/v1/models/atoms.py (hunk 15 above)
Comment (kave) [inline cfgov/v1/models/atoms.py]: thats one, i can make that change

File: cfgov/v1/models/molecules.py [hunk 17]
```diff
@@ -12,121 +12,90 @@ def isRequired(field_name):
...
+        template = '_includes/molecules/image-text-25-75.html'
 
 
 class TextIntroduction(blocks.StructBlock):
     heading = blocks.CharBlock(max_length=100, required=True)
     intro = blocks.CharBlock(max_length=100, required=True)
     body = blocks.RichTextBlock(required=False)
-    link_url = blocks.URLBlock(required=False)
-    link_text = blocks.CharBlock(max_length=100, required=False)
+    link = atoms.Hyperlink()
```
Comment (KimberlyMunoz) [inline cfgov/v1/models/molecules.py]: Can we make this link not-required?

image

File: cfgov/jinja2/v1/_includes/molecules/form-field-with-button.html (hunk 12 above)
Comment (kurtw) [inline cfgov/jinja2/v1/_includes/molecules/form-field-with-button.html]: So do we need separate fields for `id` and `name`? or are they the same info

File: cfgov/v1/models/molecules.py (hunk 14 above)
Comment (kurtw) [inline cfgov/v1/models/molecules.py]: In the templates, there's a variable called `header`. Could this be what the paragraph text atom was supposed to be? @KimberlyMunoz https://github.com/cfpb/cfgov-refresh/pull/1154/files#diff-3f1f0f338f0a28057bf1dac682d9dca2R31

File: cfgov/v1/models/organisms.py [hunk 18]
```diff
@@ -1,4 +1,8 @@
 from wagtail.wagtailcore import blocks
+from wagtail.wagtailcore.fields import RichTextField, StreamField
+from wagtail.wagtailadmin.edit_handlers import FieldPanel, StreamFieldPanel
+
+from . import molecules
 
 
 class Well(blocks.StructBlock):
```
Comment (KimberlyMunoz) [inline cfgov/v1/models/organisms.py]: The well right now should just be one rich text field. The macro originally had a template path for demo-ing purposes, but content should just see that rich text field.

image

File: cfgov/v1/models/molecules.py (hunk 17 above)
Comment (kave) [inline cfgov/v1/models/molecules.py]: sure i can do that

File: cfgov/v1/models/organisms.py (hunk 18 above)
Comment (kave) [inline cfgov/v1/models/organisms.py]: @KimberlyMunoz is this the same for Full Width Text Copy?

File: cfgov/v1/models/molecules.py [hunk 19]
```diff
@@ -12,121 +12,90 @@ def isRequired(field_name):
...
 class ImageText2575(blocks.StructBlock):
     heading = blocks.CharBlock(max_length=100, required=True)
     body = blocks.RichTextBlock(required=True)
-    image = ImageChooserBlock(required=False)
-    image_path = blocks.CharBlock(required=False)
-    image_alt = blocks.CharBlock(required=False)
-    link_url = blocks.URLBlock(required=False)
-    link_text = blocks.CharBlock(max_length=100, required=False)
+    image = atoms.ImageBasic()
+    link = atoms.Hyperlink()
```
Comment (KimberlyMunoz) [inline cfgov/v1/models/molecules.py]: Can we make this link optional?

File: cfgov/v1/models/molecules.py [hunk 20]
```diff
@@ -12,121 +12,90 @@ def isRequired(field_name):
...
 
 class ImageText2575(blocks.StructBlock):
     heading = blocks.CharBlock(max_length=100, required=True)
     body = blocks.RichTextBlock(required=True)
-    image = ImageChooserBlock(required=False)
-    image_path = blocks.CharBlock(required=False)
-    image_alt = blocks.CharBlock(required=False)
-    link_url = blocks.URLBlock(required=False)
-    link_text = blocks.CharBlock(max_length=100, required=False)
+    image = atoms.ImageBasic()
```
Comment (KimberlyMunoz) [inline cfgov/v1/models/molecules.py]: Can we drop the url field and leave only the Chooser and the Alt text field?

image

File: cfgov/v1/models/molecules.py (hunk 19 above)
Comment (kave) [inline cfgov/v1/models/molecules.py]: will do

File: cfgov/jinja2/v1/_includes/molecules/hero.html (hunk 16 above)
Comment (kave) [inline cfgov/jinja2/v1/_includes/molecules/hero.html]: done

File: cfgov/v1/models/molecules.py (hunk 14 above)
Comment (kave) [inline cfgov/v1/models/molecules.py]: these field names will be resolved in a separate PR https://gist.github.com/sebworks/5586cfa6fe01090804df

File: cfgov/v1/models/molecules.py (hunk 20 above)
Comment (kave) [inline cfgov/v1/models/molecules.py]: done

File: cfgov/v1/models/organisms.py (hunk 18 above)
Comment (KimberlyMunoz) [inline cfgov/v1/models/organisms.py]: Yep.

File: cfgov/jinja2/v1/_includes/molecules/form-field-with-button.html (hunk 12 above)
Comment (kave) [inline cfgov/jinja2/v1/_includes/molecules/form-field-with-button.html]: yea good question @KimberlyMunoz

File: cfgov/v1/models/molecules.py (hunk 5 above)
Comment (kave) [inline cfgov/v1/models/molecules.py]: A dropdown could work

File: cfgov/v1/models/molecules.py [hunk 21]
```diff
@@ -12,121 +12,97 @@ def isRequired(field_name):
...
+        ('text', 'Text'),
+        ('checkbox', 'Checkbox'),
+        ('email', 'Email'),
+        ('number', 'Number'),
+        ('url', 'URL'),
+        ('radio', 'Radio'),
+    ], icon='cup', required=True)
+    name = blocks.CharBlock(max_length=100, required=False)
+    placeholder = blocks.CharBlock(max_length=100, required=False)
+    attributes = blocks.CharBlock(max_length=100, required=False)
```
Comment (KimberlyMunoz) [inline cfgov/v1/models/molecules.py]: Other than required, which we have a checkbox for, are there any other attributes we need to account for, or can we remove this? @sebworks ?

File: cfgov/v1/models/molecules.py [hunk 22]
```diff
@@ -12,121 +12,97 @@ def isRequired(field_name):
...
+        icon = 'image'
+        template = '_includes/molecules/hero.html'
+
+
+class FormFieldWithButton(blocks.StructBlock):
+    btn_text = blocks.CharBlock(max_length=100, required=True)
+
+    required = blocks.BooleanBlock(required=False)
+    id = blocks.CharBlock(max_length=100, required=False)
+    info = blocks.RichTextBlock(required=False)
```
Comment (KimberlyMunoz) [inline cfgov/v1/models/molecules.py]: Can we change this label to "disclaimer" so that folks know what block of text that is?

File: cfgov/v1/models/molecules.py [hunk 23]
```diff
@@ -12,121 +12,97 @@ def isRequired(field_name):
...
+    label = blocks.CharBlock(max_length=100, required=True)
+    type = blocks.ChoiceBlock(choices=[
+        ('text', 'Text'),
+        ('checkbox', 'Checkbox'),
+        ('email', 'Email'),
+        ('number', 'Number'),
+        ('url', 'URL'),
+        ('radio', 'Radio'),
+    ], icon='cup', required=True)
+    name = blocks.CharBlock(max_length=100, required=False)
```
Comment (KimberlyMunoz) [inline cfgov/v1/models/molecules.py]: Can the name match the ID to reduce the need for two fields? I could easily see this getting changed and breaking things.

File: cfgov/v1/models/molecules.py [hunk 24]
```diff
@@ -12,121 +12,97 @@ def isRequired(field_name):
...
+
+
+class FormFieldWithButton(blocks.StructBlock):
+    btn_text = blocks.CharBlock(max_length=100, required=True)
+
+    required = blocks.BooleanBlock(required=False)
+    id = blocks.CharBlock(max_length=100, required=False)
+    info = blocks.RichTextBlock(required=False)
+    label = blocks.CharBlock(max_length=100, required=True)
+    type = blocks.ChoiceBlock(choices=[
```
Comment (KimberlyMunoz) [inline cfgov/v1/models/molecules.py]: When I tested this I couldn't get the dropdown to appear.

**Comment by KimberlyMunoz** [comment]
Great changes on the groupings with the title! I assume those changes mean that the order of those organisms/molecules is now fixed. If so, can we put them in this order from the spec?

- 50-50 molecule
- 25-75 molecule
- Grey well
- Half link blob

image

File: cfgov/v1/models/molecules.py (hunk 24 above)
Comment (KimberlyMunoz) [inline cfgov/v1/models/molecules.py]: Pulled in the recent changes, and this is good now!

**Comment by KimberlyMunoz** [comment]
I can't get the grey well organism to appear on the page.

**Comment by KimberlyMunoz** [comment]
Can we have the markup for containers around the different molecules only appear if the molecule is in use? In this case, I don't have any molecules other than the half-link blob, but there's lots of extra spacing from `block` and headings. 

image

**Comment by KimberlyMunoz** [comment]
There's some feedback from Content on changing the names of some of the fields to make it clearer for people entering content. 

- [ ] Change "Is widescreen" to "Use 16:9 image"
- [ ] Change "is button" to "Show link as button"
- [ ] Enable the ability for multiple links with the 50-50 image. (Also an issue for 75-25 text-image)

File: cfgov/v1/models/molecules.py (hunk 24 above)
Comment (kave) [inline cfgov/v1/models/molecules.py]: :+1:

File: cfgov/v1/models/molecules.py (hunk 21 above)
Comment (sebworks) [inline cfgov/v1/models/molecules.py]: yes, we just need the `required` setting.

File: cfgov/v1/models/molecules.py (hunk 22 above)
Comment (kave) [inline cfgov/v1/models/molecules.py]: donzo

**Comment by kave** [comment]
@KimberlyMunoz 
- reorganized the groupings in the landing page as you mentioned.
- fixed Well
- Resolved content feedback but i need you to verify the styling on the templates since its now rendering multiple links instead of 1

**Comment by kave** [comment]
@KimberlyMunoz For your markup for containers question. In what scenario did you see this happen? trying to duplicate locally

File: cfgov/v1/models/molecules.py (hunk 23 above)
Comment (kave) [inline cfgov/v1/models/molecules.py]: @KimberlyMunoz sure but is there a need for both fields? couldn't we just only use 1 on the templates and get rid of the other?

File: cfgov/v1/models/molecules.py (hunk 21 above)
Comment (kave) [inline cfgov/v1/models/molecules.py]: its been removed in the latest push and templates have been updated

**Comment by KimberlyMunoz** [comment]
For the container thing, I replicated by creating a group of half-link blobs and getting extra blank space.

The styling on multiple links looks great, but is it possible to make the multiple links optional? There's going to be circumstances like on the `/the-bureau/` page where links won't be needed.

I still can't get the grey well to appear.

**Comment by tnferrell** [comment]
The only things standing out to me are...

It might be nice to have that multiple link feature you added to the Half-Width Blob on the 25-75 Text Image Molecule and 50-50 Image Text. I just imagine someday we'll run into a use case where that's desired and having the flexibility would be nice. 

These are probably more Wagtail interface questions...but will we be able to format text within these input fields? In the screenshots you sent @KimberlyMunoz, I didn't see any WYSIWYG buttons for bolding, italicizing, etc, etc.  The input fields were also small, like the introduction text field in the text introduction molecule. Do they expand on input or does a scrollbar appear? Just thinking about how easy it'll be to proofread and update in the interface.

**Comment by KimberlyMunoz** [comment]
> It might be nice to have that multiple link feature you added to the Half-Width Blob on the 25-75 Text Image Molecule and 50-50 Image Text. I just imagine someday we'll run into a use case where that's desired and having the flexibility would be nice.

Done!

>These are probably more Wagtail interface questions...but will we be able to format text within these input fields?

In the thick fields they appear when you are in those fields.

**Comment by tnferrell** [comment]
@KimberlyMunoz So speedy. 

Cool and cool. Do you know if the input fields expand when content is entered or is it a scrollbar situation? Or is this a question for the BEWDs?

**Comment by KimberlyMunoz** [comment]
Like this when you click on the field.

image

**Comment by schaferjh** [comment]
All of this is looking great to me. :fireworks: 

One very small comment: For the **text introduction** only the heading field is absolutely required. That way we can use this element for pages that just have a title (Contact Us) or pages that have a title, and an intro and some body copy and a link. From what I can tell at the moment only the link is optional -- but I may be misreading things!

**Comment by KimberlyMunoz** [comment]
It looks like the Intro Paragraph and Title are marked as required, the body paragraph and link aren't required.

**Comment by kave** [comment]
Yup i can make the change

**Comment by kave** [comment]
@KimberlyMunoz Fixed the extra block space with if cases and fixed well for landing page (properly this time lol) and updated the required fields for text intro

**Comment by jimmynotjim** [comment]
Justin just updated the Intro Specs to requiring the title and body text, but not the intro or link. We should nail this one down so we don't have to keep adjusting it.

**Comment by duelj** [comment]
Sorry thing I misspoke, I would follow what Jess has said. My apologies.

**Comment by kave** [comment]
PSA: ignore all the migrations scripts i'll be merging them into 1 file at the end of this PR

File: cfgov/jinja2/v1/_includes/molecules/form-field-with-button.html (hunk 12 above)
Comment (kave) [inline cfgov/jinja2/v1/_includes/molecules/form-field-with-button.html]: Resolved by removing name field so only Id is required. Also added id validation to only allow alphanumberic values along with `-` `_`

**Comment by KimberlyMunoz** [comment]
giphy

:+1:

**Comment by kave** [comment]
:dancer:
